openbas-api/src/main/java/io/openbas/fulltextsearch/FullTextSearchService.java

```diff
@@ -0,0 +1,216 @@
+          (Class<T>) User.class, new FullTextSearchCountResult(User.class.getSimpleName(), 0L),
+          (Class<T>) Team.class, new FullTextSearchCountResult(Team.class.getSimpleName(), 0L),
+          (Class<T>) Organization.class, new FullTextSearchCountResult(Organization.class.getSimpleName(), 0L),
+          (Class<T>) Scenario.class, new FullTextSearchCountResult(Scenario.class.getSimpleName(), 0L),
+          (Class<T>) Exercise.class, new FullTextSearchCountResult(Exercise.class.getSimpleName(), 0L)
+      );
+    }
+
+    Map<Class<T>, FullTextSearchCountResult> results = new HashMap<>();
+    String finalSearchTerm = Arrays.stream(searchTerm.split(" "))
+        .map((s) -> "(" + s + ":*)")
+        .collect(Collectors.joining(" & "));
+
+    // Search on assets
+    long assets = this.assetRepository.count(SpecificationUtils.fullTextSearch(finalSearchTerm, "name"));
+    results.put((Class<T>) Asset.class, new FullTextSearchCountResult(Asset.class.getSimpleName(), assets));
+
+    // Search on asset groups
+    long assetGroups = this.assetGroupRepository.count(
+        SpecificationUtils.fullTextSearch(finalSearchTerm, "name"));
+    results.put((Class<T>) AssetGroup.class, new FullTextSearchCountResult(AssetGroup.class.getSimpleName(), assetGroups));
+
+    // Search on users
+    long users = this.userRepository.count(SpecificationUtils.fullTextSearch(finalSearchTerm, "email"));
+    results.put((Class<T>) User.class, new FullTextSearchCountResult(User.class.getSimpleName(), users));
+
+    // Search on teams
+    long teams = this.teamRepository.count(SpecificationUtils.fullTextSearch(finalSearchTerm, "name"));
+    results.put((Class<T>) Team.class, new FullTextSearchCountResult(Team.class.getSimpleName(), teams));
+
+    // Search on organizations
+    long organizations = this.organizationRepository.count(
+        SpecificationUtils.fullTextSearch(finalSearchTerm, "name"));
+    results.put((Class<T>) Organization.class, new FullTextSearchCountResult(Organization.class.getSimpleName(), organizations));
+
+    // Search on scenarios
+    long scenarios = this.scenarioRepository.count(
+        SpecificationUtils.fullTextSearch(finalSearchTerm, "name"));
+    results.put((Class<T>) Scenario.class, new FullTextSearchCountResult(Scenario.class.getSimpleName(), scenarios));
+
+    // Search on simulations
+    long exercises = this.exerciseRepository.count(
+        SpecificationUtils.fullTextSearch(finalSearchTerm, "name"));
+    results.put((Class<T>) Exercise.class, new FullTextSearchCountResult(Exercise.class.getSimpleName(), exercises));
+
+    return results;
+  }
+
+  @AllArgsConstructor
+  @Data
+  public static class FullTextSearchCountResult {
+
+    @NotBlank
+    private String clazz;
+    @NotBlank
+    private long count;
+
+  }
+
+  @Data
+  public static class FullTextSearchResult {
+
+    @NotBlank
+    private String id;
+    @NotBlank
+    private String name;
+    @NotBlank
+    private String clazz;
+
```

Comment:
Maybe add some additional information for differentiation? tags, short description?

Reply:
Good point, I will add some infos